### ISSUE TITLE
Validation suddenly fails

### ISSUE THREAD

**Issue by lunactic** [common-workflow-language/cwltool#861]
## Expected Behavior
The validation of the echo.cwl test job executes successfully.

## Actual Behavior
The validation fails with the error:

```
/usr/bin/cwltool 1.0.20180808165120
Resolved 'tests/echo.cwl' to 'file:///home/lunactic/DEV/cwltool_orig/tests/echo.cwl'
I'm sorry, I couldn't load this CWL file, try again with --debug for more information.
The error was: workflowjobj must be a dict, got '<class 'ruamel.yaml.comments.CommentedMap'>': ordereddict([('cwlVersion', 'v1.0'), ('class', 'CommandLineTool'), ('inputs', [ordereddict([('id', 'inp'), ('type', 'string'), ('inputBinding', ordereddict())])]), ('outputs', [ordereddict([('id', 'out'), ('type', 'string'), ('outputBinding', ordereddict([('glob', 'out.txt'), ('loadContents', True), ('outputEval', '$(self[0].contents)')]))])]), ('baseCommand', 'echo'), ('stdout', 'out.txt'), ('id', 'file:///home/lunactic/DEV/cwltool_orig/tests/echo.cwl')])
```

## Workflow Code
```
cwtool --validate tests/echo.cwl
```

## Full Traceback
```pytb
/usr/bin/cwltool 1.0.20180808165120
Resolved 'tests/echo.cwl' to 'file:///home/lunactic/DEV/cwltool_orig/tests/echo.cwl'
I'm sorry, I couldn't load this CWL file, try again with --debug for more information.
The error was: workflowjobj must be a dict, got '<class 'ruamel.yaml.comments.CommentedMap'>': ordereddict([('cwlVersion', 'v1.0'), ('class', 'CommandLineTool'), ('inputs', [ordereddict([('id', 'inp'), ('type', 'string'), ('inputBinding', ordereddict())])]), ('outputs', [ordereddict([('id', 'out'), ('type', 'string'), ('outputBinding', ordereddict([('glob', 'out.txt'), ('loadContents', True), ('outputEval', '$(self[0].contents)')]))])]), ('baseCommand', 'echo'), ('stdout', 'out.txt'), ('id', 'file:///home/lunactic/DEV/cwltool_orig/tests/echo.cwl')])
```

## Your Environment
* cwltool version: 1.0.20180808165120
* python version: 3.7 (tested with 3.5 as well)

**Comment by kellrott** [comment]
I'm getting the same error. 
```
 workflowjobj must be a dict, got '<class 'ruamel.yaml.comments.CommentedMap'>':
```

**Comment by tetron** [comment]
Could you run `cwtool --debug --validate tests/echo.cwl` ?

**Comment by tetron** [comment]
ruamel.yaml 0.15.52 released changed it so that `CommentedMap` no longer inherits from `ordereddict`.  This breaks backwards compatibility in multiple ways, code in cwltool assumed `CommentedMap` could be used as a drop-in replacement for `dict` objects, but that is no longer entirely true.   

Working on fixes:

https://github.com/common-workflow-language/schema_salad/pull/195

**Comment by mr-c** [comment]
Fixed in #864
